### ISSUE TITLE
It is not possible to remove servicebinding with `bindAsFiles` 

### ISSUE THREAD

**Issue by wtrocki** [redhat-developer/service-binding-operator#890]
The operator looks for the wrong secret and staring to crash. 
Only one way to remove it is to remove finalized.

Then this causes all future bindings to fail.

**Comment by DhritiShikhar** [comment]
I was able to reproduce the scenario. 

This is the ServiceBinding I created with `bindAsFiles` flag set:

```
➜  service-binding-operator-yamls git:(master) oc get sbr sb-1 -o yaml
apiVersion: binding.operators.coreos.com/v1alpha1
kind: ServiceBinding
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"binding.operators.coreos.com/v1alpha1","kind":"ServiceBinding","metadata":{"annotations":{},"name":"sb-1","namespace":"sb-1"},"spec":{"application":{"group":"apps","name":"nodejs-rest-http-crud","resource":"deployments","version":"v1"},"bindAsFiles":true,"services":[{"group":"postgresql.baiju.dev","kind":"Database","name":"db-demo","version":"v1alpha1"}]}}
  creationTimestamp: "2021-03-18T11:33:24Z"
  finalizers:
  - finalizer.servicebinding.openshift.io
  generation: 1
  managedFields:
  - apiVersion: binding.operators.coreos.com/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:kubectl.kubernetes.io/last-applied-configuration: {}
      f:spec:
        .: {}
        f:application:
          .: {}
          f:group: {}
          f:name: {}
          f:resource: {}
          f:version: {}
        f:bindAsFiles: {}
        f:services: {}
    manager: kubectl-client-side-apply
    operation: Update
    time: "2021-03-18T11:33:24Z"
  - apiVersion: binding.operators.coreos.com/v1alpha1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .: {}
          v:"finalizer.servicebinding.openshift.io": {}
      f:spec:
        f:application:
          f:bindingPath:
            .: {}
            f:containersPath: {}
            f:secretPath: {}
          f:labelSelector: {}
        f:detectBindingResources: {}
      f:status:
        .: {}
        f:conditions:
          .: {}
          k:{"type":"CollectionReady"}:
            .: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
          k:{"type":"InjectionReady"}:
            .: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
          k:{"type":"Ready"}:
            .: {}
            f:lastTransitionTime: {}
            f:message: {}
            f:reason: {}
            f:status: {}
            f:type: {}
        f:secret: {}
    manager: manager
    operation: Update
    time: "2021-03-18T11:33:27Z"
  name: sb-1
  namespace: sb-1
  resourceVersion: "119769"
  selfLink: /apis/binding.operators.coreos.com/v1alpha1/namespaces/sb-1/servicebindings/sb-1
  uid: e76bdfc1-d0aa-4646-85a4-0f92dd916456
spec:
  application:
    group: apps
    name: nodejs-rest-http-crud
    resource: deployments
    version: v1
  bindAsFiles: true
  services:
  - group: postgresql.baiju.dev
    kind: Database
    name: db-demo
    version: v1alpha1
status:
  conditions:
  - lastTransitionTime: "2021-03-18T11:33:24Z"
    message: ""
    reason: BindingInjected
    status: "True"
    type: CollectionReady
  - lastTransitionTime: "2021-03-18T11:33:24Z"
    message: ""
    reason: BindingInjected
    status: "True"
    type: InjectionReady
  - lastTransitionTime: "2021-03-18T11:33:24Z"
    message: ""
    reason: BindingInjected
    status: "True"
    type: Ready
  secret: sb-1-15619bed77c9fc2ae0e7132438bc70bf790f1a1c
```


On deleting the ServiceBinding, though the message says `sb-1` is deleted, the command never completes execution and is stuck.

```
➜  service-binding-operator-yamls git:(master) ✗ oc delete sbr sb-1                                        
servicebinding.binding.operators.coreos.com "sb-1" deleted

```

**Comment by pedjak** [comment]
@wtrocki can you please check if the issue is fixed on the latest master, after #919 was merged?

**Comment by wtrocki** [comment]
Fixed
